plugins/woocommerce/tests/php/src/Blocks/Templates/SingleProductTemplateTests.php

```diff
@@ -11,6 +11,172 @@
+		);
+		$result_without_whitespace                           = preg_replace( '/\s+/', '', $result[0]->content );
+
+		$this->assertEquals(
+			$expected_single_product_template_without_whitespace,
+			$result_without_whitespace
+		);
+	}
+
+	/**
+	 * Test that the Single Product template content isn't updated if it
+	 * contains a pattern with the Legacy Template block.
+	 *
+	 */
+	public function test_dont_update_single_product_content_with_legacy_template_inside_a_pattern() {
+		register_block_pattern(
+			'test-pattern',
+			array(
+				'title'       => 'Test Pattern',
+				'description' => 'Test Pattern Description',
+				'content'     => '<!-- wp:woocommerce/legacy-template {"template":"single-product"} /-->',
+			)
+		);
+		$single_product_template                 = new SingleProductTemplate();
+		$default_single_product_template_content = '
+			<!-- wp:template-part {"slug":"header","theme":"twentytwentythree","tagName":"header"} /-->
+			<!-- wp:pattern {"slug":"test-pattern"} /-->
+			<!-- wp:template-part {"slug":"footer","theme":"twentytwentythree","tagName":"footer"} /-->';
+
+		$template          = new \WP_Block_Template();
+		$template->slug    = 'single-product';
+		$template->title   = 'Single Product';
+		$template->content = $default_single_product_template_content;
+		$template->type    = 'wp_template';
+
+		$result = $single_product_template->update_single_product_content(
+			array(
+				$template,
+			),
+		);
+
+		$this->assertEquals(
+			$default_single_product_template_content,
+			$result[0]->content
+		);
+	}
+
+	/**
+	 * Test that the Single Product template content is updated if it doesn't
+	 * contain the Legacy Template block.
+	 *
+	 */
+	public function test_update_single_product_content_with_legacy_template_inside_a_pattern() {
+		register_block_pattern(
+			'test-pattern',
+			array(
+				'title'       => 'Test Pattern',
+				'description' => 'Test Pattern Description',
+				'content'     => '<!-- wp:woocommerce/product-image-gallery /-->',
+			)
+		);
+		$single_product_template                  = new SingleProductTemplate();
+		$default_single_product_template_content  = '
+			<!-- wp:template-part {"slug":"header","theme":"twentytwentythree","tagName":"header"} /-->
+			<!-- wp:pattern {"slug":"test-pattern"} /-->
+			<!-- wp:template-part {"slug":"footer","theme":"twentytwentythree","tagName":"footer"} /-->';
+		$expected_single_product_template_content = '
+			<!-- wp:template-part {"slug":"header","theme":"twentytwentythree","tagName":"header"} /-->
+			<!-- wp:group {"className":"woocommerce product","__wooCommerceIsFirstBlock":true,"__wooCommerceIsLastBlock":true} -->
+			<div class="wp-block-group woocommerce product">
+			<!-- wp:pattern {"slug":"test-pattern"} /-->
+			</div>
+			<!-- /wp:group -->
+			<!-- wp:template-part {"slug":"footer","theme":"twentytwentythree","tagName":"footer"} /-->';
+
+		$template          = new \WP_Block_Template();
+		$template->slug    = 'single-product';
```

Comment:
Considering we have logic that looks for this slug, should we test conditions with a different slug?

Reply:
I added an extra test in 49c140c7f7fe05b69eef34d67043e0f8ad647496. Please let me know if that's what you had in mind.

Reply:
Perfect thanks!